openstudiocore/CMakeLists.txt

```diff
@@ -1032,44 +1014,13 @@ add_definitions(-DBOOST_CHRONO_HEADER_ONLY)
-  set(CPPRESTSDK_ZIP "cpprestsdk-2.10.8-Linux-x86_64.tar.xz")
-  set(CPPRESTSDK_DIR "cpprestsdk-2.10.8-Linux-x86_64")
-  set(CPPRESTSDK_EXPECTED_HASH "866e8fa7ce1e507fcb19b0ab9b89399c")
-  if(EXISTS "${CMAKE_BINARY_DIR}/${CPPRESTSDK_ZIP}")
-    file(MD5 "${CMAKE_BINARY_DIR}/${CPPRESTSDK_ZIP}" CPPRESTSDK_HASH)
-  endif()
-  if(NOT EXISTS "${CMAKE_BINARY_DIR}/${CPPRESTSDK_ZIP}" OR NOT EXISTS "${CMAKE_BINARY_DIR}/${CPPRESTSDK_DIR}" OR NOT "${CPPRESTSDK_HASH}" MATCHES "${CPPRESTSDK_EXPECTED_HASH}")
-    message(STATUS "Downloading cpprestsdk)")
-    file(DOWNLOAD "http://openstudio-resources.s3.amazonaws.com/dependencies/${CPPRESTSDK_ZIP}" "${CMAKE_BINARY_DIR}/${CPPRESTSDK_ZIP}" TIMEOUT 120 INACTIVITY_TIMEOUT 120 SHOW_PROGRESS EXPECTED_MD5 ${CPPRESTSDK_EXPECTED_HASH})
-    execute_process(COMMAND ${CMAKE_COMMAND} -E remove_directory "${CMAKE_BINARY_DIR}/${CPPRESTSDK_DIR}")
-    execute_process(COMMAND ${CMAKE_COMMAND} -E tar xfz "${CMAKE_BINARY_DIR}/${CPPRESTSDK_ZIP}" WORKING_DIRECTORY "${CMAKE_BINARY_DIR}")
-  endif()
-  #find_package(cpprestsdk REQUIRED PATHS "${CMAKE_BINARY_DIR}/${CPPRESTSDK_DIR}/lib/cpprestsdk")
-  set(CPPRESTSDK_INCLUDE_DIR "${CMAKE_BINARY_DIR}/${CPPRESTSDK_DIR}/include")
-  include_directories(${CPPRESTSDK_INCLUDE_DIR})
-  set(CPPRESTSDK_LIB "${CMAKE_BINARY_DIR}/${CPPRESTSDK_DIR}/lib/libcpprest.a")
-elseif(WIN32)
-  set(CPPRESTSDK_ZIP "cpprestsdk-2.10.8-win64.zip")
-  set(CPPRESTSDK_DIR "cpprestsdk-2.10.8-win64")
-  set(CPPRESTSDK_EXPECTED_HASH "467df76fa7c96fc3465a4b5f596361f8")
-  if(EXISTS "${CMAKE_BINARY_DIR}/${CPPRESTSDK_ZIP}")
-    file(MD5 "${CMAKE_BINARY_DIR}/${CPPRESTSDK_ZIP}" CPPRESTSDK_HASH)
-  endif()
-  if(NOT EXISTS "${CMAKE_BINARY_DIR}/${CPPRESTSDK_ZIP}" OR NOT EXISTS "${CMAKE_BINARY_DIR}/${CPPRESTSDK_DIR}" OR NOT "${CPPRESTSDK_HASH}" MATCHES "${CPPRESTSDK_EXPECTED_HASH}")
-    message(STATUS "Downloading cpprestsdk)")
-    file(DOWNLOAD "http://openstudio-resources.s3.amazonaws.com/dependencies/${CPPRESTSDK_ZIP}" "${CMAKE_BINARY_DIR}/${CPPRESTSDK_ZIP}" TIMEOUT 120 INACTIVITY_TIMEOUT 120 SHOW_PROGRESS EXPECTED_MD5 ${CPPRESTSDK_EXPECTED_HASH})
-    execute_process(COMMAND ${CMAKE_COMMAND} -E remove_directory "${CMAKE_BINARY_DIR}/${CPPRESTSDK_DIR}")
-    execute_process(COMMAND ${CMAKE_COMMAND} -E tar xfz "${CMAKE_BINARY_DIR}/${CPPRESTSDK_ZIP}" WORKING_DIRECTORY "${CMAKE_BINARY_DIR}")
-  endif()
-  #find_package(cpprestsdk REQUIRED PATHS "${CMAKE_BINARY_DIR}/${CPPRESTSDK_DIR}/lib/cpprestsdk")
-  set(CPPRESTSDK_INCLUDE_DIR "${CMAKE_BINARY_DIR}/${CPPRESTSDK_DIR}/include")
-  include_directories(${CPPRESTSDK_INCLUDE_DIR})
-  set(CPPRESTSDK_LIB_RELEASE "${CMAKE_BINARY_DIR}/${CPPRESTSDK_DIR}/lib/cpprest141_2_10.lib")
-  set(CPPRESTSDK_LIB_DEBUG "${CMAKE_BINARY_DIR}/${CPPRESTSDK_DIR}/lib/cpprest141_2_10d.lib")
+if(WIN32)
+  # cpprest defines its own NOMINMAX and this is causes a macro redefinition "warning as error"
+  remove_definitions(-DNOMINMAX)
+endif()
+add_subdirectory(../cpprestsdk cpprestsdk EXCLUDE_FROM_ALL)
```

Comment:
Does this build correctly with the OpenSSL dependency from Conan?  Why is it excluded from all?  Isn't cpprestsdk a dependency of openstudio_utilities?

Reply:
I think @lefticus has it right here. just because it is excluded from the "ALL" target doesn't prevent someone like openstudio_utilities from putting a dependency on it.